src/Strategy/Puli.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+use Http\Discovery\Exception\PuliUnavailableException;
+
+/**
+ * @author David de Boer <david@ddeboer.nl>
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+class Puli implements DiscoveryStrategy
+{
+    /**
+     * @var \Puli\GeneratedPuliFactory
+     */
+    private static $puliFactory;
+
+    /**
+     * @var \Puli\Discovery\Api\Discovery
```

Comment:
I did not add use statements for one (silly) reason: When Puli is not installed my IDE will mark this class "with errors" because the class in the use statement does not exist.

I'll add them if anyone wants them.